lib/saml2.coffee

```diff
@@ -477,13 +479,13 @@ module.exports.ServiceProvider =
     #
     # Rest of options can be set/overwritten by the identity provider and/or at function call.
     constructor: (options) ->
-      {@entity_id, @private_key, @certificate, @assert_endpoint, @alt_private_keys, @alt_certs} = options
+      {@entity_id, @private_key, @certificate, @assert_endpoint, @alt_private_keys, @alt_certs, @require_session_index} = options
```

Comment:
I think it would be more clear to move the defaulting of session index to true to this constructor, instead of passing the null value through to `get_session_index` - seems like there would be a high potential for bugs if we decided to use it somewhere else and didn't remember to add the default there too.